tests/test-db.js

```diff
@@ -0,0 +1,72 @@
+"use strict";
+
+const test = require("tape-async");
+
+const DB = require("../db/DB");
+const getSha1 = require("../sha1-utils");
+
+test("getSubscriberByEmailAndToken accepts email and token and returns subscriber", async t => {
+  const testEmail = "unverifiedemail@test.com";
+  const testToken = "0e2cb147-2041-4e5b-8ca9-494e773b2cf0";
+  const subscriber = await DB.getSubscriberByEmailAndToken(testEmail, testToken);
+
+  t.ok(subscriber.email === testEmail);
+  t.ok(subscriber.verification_token === testToken);
+});
+
+test("getSubscribersByHashes accepts hashes and only returns verified subscribers", async t => {
+  const testHashes = [
+    "firefoxaccount@test.com",
+    "unverifiedemail@test.com",
+    "verifiedemail@test.com",
+  ].map(email => getSha1(email));
+  const subscribers = await DB.getSubscribersByHashes(testHashes);
+  for (const subscriber of subscribers) {
+    t.ok(subscriber.verified);
+  }
+});
+
+test("addSubscriberUnverifiedEmailHash accepts email and returns unverified subscriber with sha1 hash and verification token", async t => {
+  const testEmail = "test@test.com";
+  // https://stackoverflow.com/a/13653180
+  const uuidRE = /^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i;
```

Comment:
Do we do UUID `verification_token` validation anywhere else in the server (or _should_ we be validating it elsewhere)?
Should this RegExp (or a helper function) be added to the ../db/DB.js file as some `isValidVerificationToken(token) {...}` method?

Reply:
Hmm ... it would be belt-and-suspenders validation, since we're checking the token equality in the database already. We can make it a follow-up.